### PR TITLE
Doctest 2.4.9

### PR DESCRIPTION
**Pull request by nlohmann**
Update to Doctest 2.4.9

(Replaces #3228)

### CODE REVIEW

**Comment by nlohmann** [comment]
There is an issue with IPCC; reported upstream: https://github.com/doctest/doctest/issues/664

File: cmake/ci.cmake [hunk 1]
```diff
@@ -928,6 +928,7 @@ add_custom_target(ci_icpc
         -DCMAKE_BUILD_TYPE=Debug -GNinja
         -DCMAKE_C_COMPILER=icc -DCMAKE_CXX_COMPILER=icpc
         -DJSON_BuildTests=ON -DJSON_FastTests=ON
+        -DCMAKE_CXX_FLAGS="-D_MSC_VER=0 -D_MSC_FULL_VER=0"
```
Comment (falbrechtskirchinger) [inline cmake/ci.cmake:931]:
```suggestion
        -DCMAKE_CXX_FLAGS="-D_MSC_VER=0;-D_MSC_FULL_VER=0"
```

Comment (nlohmann) [reply cmake/ci.cmake:931]: Thanks for the hint. But since this was a only a hack in the first place, I'd undo it and wait for a fix from upstream.

Comment (falbrechtskirchinger) [reply cmake/ci.cmake:931]: Just a reminder to use semicolons when dealing with CMake lists in the future. :-)

Comment (nlohmann) [reply cmake/ci.cmake:931]: This was not the first one, and I fear it was also not the last one :-)

**Review by shaischaudhry — APPROVED** [review]

**Pull request closed without merging**